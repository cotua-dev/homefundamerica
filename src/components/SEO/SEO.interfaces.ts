interface Meta
{
    name?: undefined;
    content: string;
    property: string;
}

export interface Props
{
    lang?: string;
    description?: string;
    meta?: ConcatArray<Meta>;
    keywords?: string[];
    title: string;
    siteUrl?: string;
}

export interface FetchSEODataQuery
{
    site: {
        siteMetadata: {
            title: string;
            description: string;
            author: string;
            siteUrl: string;
        };
    };
}

export interface PureSEOComponentData extends Props
{
    data?: FetchSEODataQuery;
}
