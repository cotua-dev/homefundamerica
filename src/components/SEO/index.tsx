import React, { FunctionComponent } from 'react';
import { Helmet } from 'react-helmet';
import {
    graphql,
    StaticQuery,
} from 'gatsby';
import {
    Props,
    FetchSEODataQuery,
    PureSEOComponentData,
} from './SEO.interfaces';

export const PureSEO: Function = ({ data, description, lang = 'en', meta = [], keywords = [], title = '' }: PureSEOComponentData): JSX.Element =>
{
    const metaDescription: string = description || (data ? data.site.siteMetadata.description : '');
    const metaUrl: string = data ? data.site.siteMetadata.siteUrl : '';

    return (
        <Helmet
            htmlAttributes={{ lang }}
            title={title}
            titleTemplate={`%s`}
            meta={[
                { name: 'description', content: metaDescription },
                { property: 'og:title', content: title },
                { property: 'og:description', content: metaDescription },
                { property: 'og:type', content: 'website' },
                { property: 'og:url', content: metaUrl },
                { name: 'twitter:card', content: 'summary_large_image' },
                { name: 'twitter:creator', content: '' },
                { name: 'twitter:title', content: title },
                { name: 'twitter:description', content: metaDescription },
                { name: "theme-color", content: "#1F72C1" },
            ].concat(
                keywords && (keywords.length > 0) ?
                { name: 'keywords', content: keywords.join(', ') } :
                []
            ).concat(((meta.length > 0) ? meta : []))}
        />
    );
}

const SEO: FunctionComponent<Props> = (props: Props): JSX.Element =>
(
    <StaticQuery
        query={graphql`
            query FetchSEOData {
                site {
                    siteMetadata {
                        title
                        description
                        author
                        siteUrl
                    }
                }
            }
        `}
        render={(data: FetchSEODataQuery): JSX.Element => <PureSEO {...props} data={data}/>}
    />
);

export default SEO;
