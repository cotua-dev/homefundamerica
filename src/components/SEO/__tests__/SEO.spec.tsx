import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { PureSEO as SEO } from '..';
import { PureSEOComponentData } from '../SEO.interfaces';

describe('SEO component', (): void => {
    test('renders', (): void => {
        const props: PureSEOComponentData = {
            lang: '',
            description: '',
            meta: [],
            keywords: [],
            title: '',
            siteUrl: '',
            data: {
                site: {
                    siteMetadata: {
                        title: '',
                        description: '',
                        author: '',
                        siteUrl: '',
                    },
                },
            },
        };
        const { container }: RenderResult = render(<SEO {...props}/>);

        expect(container.firstChild).toMatchSnapshot();
    });
});
