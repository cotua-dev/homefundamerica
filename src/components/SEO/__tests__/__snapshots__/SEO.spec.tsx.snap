// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO component renders 1`] = `null`;
